node/full_node_integration_test.go

```diff
@@ -169,25 +172,49 @@ func TestLazyAggregator(t *testing.T) {
 
 	require.NoError(err)
 
-	// delay to ensure it waits for block 1 to be built
-	time.Sleep(1 * time.Second)
+	// wait for 1st block to be built
+	require.NoError(testutils.Retry(300, 100*time.Millisecond, func() error {
```

Comment:
Looking at these checks in a row we can dry this up. 

if the test is controlling the blocks such that no blocks are produced unless `BroadcastTxCommit` is called then we can do the following

```go
	verifyHeight = func(h unit64) error {
		return testutils.Retry(300, 100*time.Millisecond, func() error {
			nodeHeight := node.(*FullNode).Store.Height()
			if nodeHeight == h {
				return nil
			}
			return fmt.Errorf("node at height %v, expected height %v", nodeHeight, h)
		})
	}
        requireNoError(verifyHeight, 1)
	_, err = client.BroadcastTxCommit(context.Background(), []byte{0, 0, 0, 1})
        requireNoError(verifyHeight, 2)
	_, err = client.BroadcastTxCommit(context.Background(), []byte{0, 0, 0, 2})
        requireNoError(verifyHeight, 3)
	_, err = client.BroadcastTxCommit(context.Background(), []byte{0, 0, 0, 3})
        requireNoError(verifyHeight, 4)
```

Reply:
as a note, I think it is a good habit to try and not use `else` statements unless absolutely necessary. I think it develops a habit of writing cleaner, dryer code.  It keeps everything more inline which makes it more readable as well 👍🏼